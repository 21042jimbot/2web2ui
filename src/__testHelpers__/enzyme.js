export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
