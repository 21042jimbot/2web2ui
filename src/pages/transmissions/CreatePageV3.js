import React from 'react';
import {
  Button,
  Page,
  Panel,
  Select,
  TextField
} from '@sparkpost/matchbox';

// with Redux
class CreatePageV3 extends React.Component {
  state = {
    recipient: '',
    template: ''
  }

  componentDidMount() {
    this.props.listTemplates();
  }

  componentDidUpdate(prevProps) {
    const { sentTo } = this.props;

    if (prevProps.sentTo.length < sentTo.length) {
      this.props.showAlert({
        message: `Sent transmission to ${sentTo[sentTo.length - 1]}!`,
        type: 'success'
      });
    }
  }

  handleRecipientChange = (event) => {
    this.setState({ recipient: event.target.value });
  }

  handleTemplateChange = (event) => {
    this.setState({ template: event.target.value });
  }

  handleSubmit = () => {
    const { recipient, template } = this.state;
    this.props.sendTransmission({ recipient, template });
  }

  render() {
    const { recipient } = this.state;
    const { sending, templates } = this.props;

    return (
      <Page
        title="New Transmission"
        primaryAction={{
          component: Button,
          content: 'Send Transmission',
          onClick: this.handleSubmit,
          disabled: Boolean(sending)
        }}
      >
        <Panel sectioned>
          <form>
            <TextField
              disabled={Boolean(sending)}
              label="Who will receive this transmission?"
              onChange={this.handleRecipientChange}
              placeholder="Enter a recipient email address"
              value={recipient}
            />
            <Select
              disabled={Boolean(sending)}
              label="What will this transmission contain?"
              onChange={this.handleTemplateChange}
              options={[
                { label: 'Select a template', value: '' },
                ...templates.map(({ id, name }) => ({ label: name, value: id }))
              ]}
            />
          </form>
        </Panel>
      </Page>
    );
  }
}

export default CreatePageV3;
