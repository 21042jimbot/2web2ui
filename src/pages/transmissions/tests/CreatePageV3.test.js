import React from 'react';
import { shallow } from 'enzyme';
import CreatePageV3 from '../CreatePageV3'; // note, not the container

describe('CreatePageV3', () => {
  const subject = (props = {}) => shallow(
    <CreatePageV3
      listTemplates={jest.fn()}
      sending={true}
      sentTo={[]}
      templates={[{ id: 'example-template', name: 'Example Template' }]}
      {...props}
    />
  );

  it('renders transmission form', () => {
    const wrapper = subject();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders disabled transmission form when sending', () => {
    const wrapper = subject({ sending: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('calls .sendTransmission onSubmit', async () => {
    const sendTransmission = jest.fn();
    const state = {
      recipient: 'me@example.com',
      template: 'example-template'
    };
    const wrapper = subject({ sendTransmission });

    wrapper.setState(state);
    wrapper.prop('primaryAction').onClick();

    expect(sendTransmission).toHaveBeenCalledWith(state);
  });
});
