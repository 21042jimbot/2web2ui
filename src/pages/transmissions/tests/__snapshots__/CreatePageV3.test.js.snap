// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePageV3 renders disabled transmission form when sending 1`] = `
<Page
  empty={Object {}}
  primaryAction={
    Object {
      "component": [Function],
      "content": "Send Transmission",
      "disabled": true,
      "onClick": [Function],
    }
  }
  title="New Transmission"
>
  <Panel
    sectioned={true}
  >
    <form>
      <TextField
        disabled={true}
        label="Who will receive this transmission?"
        onChange={[Function]}
        placeholder="Enter a recipient email address"
        resize="both"
        type="text"
        value=""
      />
      <Select
        disabled={true}
        label="What will this transmission contain?"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Select a template",
              "value": "",
            },
            Object {
              "label": "Example Template",
              "value": "example-template",
            },
          ]
        }
      />
    </form>
  </Panel>
</Page>
`;

exports[`CreatePageV3 renders transmission form 1`] = `
<Page
  empty={Object {}}
  primaryAction={
    Object {
      "component": [Function],
      "content": "Send Transmission",
      "disabled": true,
      "onClick": [Function],
    }
  }
  title="New Transmission"
>
  <Panel
    sectioned={true}
  >
    <form>
      <TextField
        disabled={true}
        label="Who will receive this transmission?"
        onChange={[Function]}
        placeholder="Enter a recipient email address"
        resize="both"
        type="text"
        value=""
      />
      <Select
        disabled={true}
        label="What will this transmission contain?"
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "Select a template",
              "value": "",
            },
            Object {
              "label": "Example Template",
              "value": "example-template",
            },
          ]
        }
      />
    </form>
  </Panel>
</Page>
`;
